corehq/apps/settings/tests/test_views.py

```diff
@@ -105,29 +107,39 @@ def _create_view_for_user(self, user):
 
 
 @override_settings(ALLOW_PHONE_AS_DEFAULT_TWO_FACTOR_DEVICE=True)
-class TwoFactorSetupView_FormKwargs_Tests(SimpleTestCase):
+class TwoFactorSetupView_RegisteredMethods_Tests(SimpleTestCase):
+
     @override_settings(ALLOW_PHONE_AS_DEFAULT_TWO_FACTOR_DEVICE=False)
-    def test_phone_methods_are_prohibited_when_settings_are_disabled(self):
+    def test_phone_methods_are_not_registered_when_settings_are_disabled(self):
         user = self._create_user(belongs_to_messaging_domain=True)
-        view = self._create_view_for_user(user)
-        self.assertFalse(view.get_form_kwargs(step='method')['allow_phone_2fa'])
+        self._create_view_for_user(user)
+        self.assertCountEqual(['generator'], [m.code for m in registry.get_methods()])
```

Comment:
Would it make sense to assert that 'sms' and 'call' were not in the registered methods, rather than 'generator' was? If 'generator' stopped being part of the results from the API, I don't think that should break this test, right?

Reply:
Great point! I ended up reverting this anyway once I started registered methods in app startup instead of the view decorator, but that is great feedback.